** Cross-species scaffolding

Generate /in silico/ mate-pair reads from single-/paired-end reads of your
organism of interest, and a closely related reference genome. Afterwards improve
your draft genome by using the scaffolding software of your choice with the
newly created read data.

**** Install / Dependencies
#+BEGIN_SRC
git clone https://github.com/thackl/cross-species-scaffolding.git

# install dependencies not on PATH to cross-species-scaffolding/util/
cd cross-species-scaffolding
make dependencies
#+END_SRC

Depends on:
- bwa, seqtk, samtools, bcftools
- seq-scripts

To use your preinstalled version, just put them into your =PATH=. To quickly
install dependencies not present in =PATH= run =make dependencies=. This will
download and install missing software into =cross-scaffolding/util/=

**** Sample
Run cross-mate on a small sample data set of two different yeast
strains. Data is downloaded automatically. Requires in =PATH=:

- curl, gzip
- sratoolkit (https://github.com/ncbi/sra-tools/wiki/Downloads)

Running the script takes about 10 minutes on 4 cores
after the download has finished.

#+BEGIN_SRC
make sample-yeast
#+END_SRC

**** Usage
Generate /in silico/ mate-pairs from single-/paired-end libraries of the target
organism and a reference assembly of a close relative. This convenience wrapper
script runs the pipeline steps: mapping, consensus calling and mate-pair
generation.

#+BEGIN_SRC
cross-mates REF.fa TARGET-LIB1_1.fq [TARGET-LIB1_2.fq TARGET-LIB2_1.fq ...]
#+END_SRC

Creates different mate-pair libraries with insert-sizes by default ranging from
500 bp to 200 kbp. Use them together with TARGET-READS to /de novo/ assemble
your organism at hand, or to further scaffold a previously generated assembly.

For power use and tailoring of the pipeline steps to computing infrastructure,
i.e. HPC cluster etc, run the different steps of the procedure
independently. Have a look at the source code of the cross-mates script for some
guidance.

**** Pipeline Workflow
***** 1. Mapping of reads of the target organism to the refernce genome using [[https://github.com/lh3/bwa][bwa mem]]/[[http://www.htslib.org/][Samtools]]:
#+BEGIN_SRC sh
# pseudo-code
bwa mem REF TARGET-READS | samtools view -bS - | samtools sort -o BAM
#+END_SRC

***** 2. Consensus computation using samtools [[http://www.htslib.org/][Samtools/BCFtools]]:
#+BEGIN_SRC sh
# pseudo-code
samtools mpileup -uf REF BAM | bcftools call -c | vcfutils.pl vcf2fq > CNS
#+END_SRC

By default, the required minimum coverage for a consensus call is 3. Regions
with lower coverage will result in N's and low quality values.

***** 3. Generation of /in-silico/ paired-end/mate-pair reads with [[https://github.com/thackl/seq-scripts][seq-frag]]:
#+BEGIN_SRC sh
# pseudo-code
seq-frag pe/mp -[silcp] ... <CNS
#+END_SRC

cross-mates runs seq-frag with default filtering, i.e. only read pairs with 0
bases below a phred score of 20 are returned. See =seq-frag --help= for details
on how to adjust the filtering if necessary.

Other cross-mates defaults for seq-frag:
- Paired-end vs. Mate-pair threshold: 1000 bp i.e. reads from inserts of sizes >= 1000 will have opposing outward orientation
- Sampling: systematic, i.e. exact insert sizes and insert fragments sampled at regularly spaced offsets
- Insert sizes: 500,1000,1500,2000,5000,10000,20000,50000,100000,200000
- Read length: 100
- Coverage: 10

**** Contact

Any kind of feedback is highly appreciated. Feel free to either report [[https://github.com/thackl/minidot/issues/new][issues]]
directly via github or shoot me an email.
